biorbd_optim/optimal_control_program.py

```diff
@@ -459,11 +461,137 @@ def add_plot(self, fig_name, update_function, phase_number=-1, **parameters):
+        # create ocp object to formulate the OCP
+        ocp = AcadosOcp()
+
+        # # set model
+        model = self.export_eocar_ode_model()
+        ocp.model = model
+
+        # set time
+        ocp.solver_options.tf = self.nlp[0]["tf"]
+
+        # set dimensions
+        ocp.dims.nx = self.nlp[0]["nx"]
+        ocp.dims.nu = self.nlp[0]["nu"]
+        ocp.dims.ny = ocp.dims.nx + ocp.dims.nu
+        ocp.dims.ny_e = self.nlp[0]["nx"]
+        ocp.dims.N = self.nlp[0]["ns"]
+
+        # set cost module
+        ocp.cost.cost_type = 'LINEAR_LS'
+        ocp.cost.cost_type_e = 'LINEAR_LS'
+
+        # set weight à modifier avec objective functions (user needs to define this TODO)
+        Q = 0.00 * np.eye(ocp.dims.nx)
+        R = 5 * np.eye(ocp.dims.nu)
+
+        ocp.cost.W = scipy.linalg.block_diag(Q, R)
+
+        ocp.cost.W_e = Q
+
+        # set Lagrange term matrices
+        ocp.cost.Vx = np.zeros((ocp.dims.ny, ocp.dims.nx))
+        ocp.cost.Vx[:ocp.dims.nx, :] = np.eye(ocp.dims.nx)
+
+        # Vu = np.zeros((ny, nu))
+        # Vu[nx:, :] = np.eye(nu)
+        # ocp.cost.Vu = Vu
+
+        Vu = np.zeros((ocp.dims.ny, ocp.dims.nu))
+        Vu[ocp.dims.nx:, :] = np.eye(ocp.dims.nu)
+        ocp.cost.Vu = Vu
+
+        # set Mayer term matrices
+        ocp.cost.Vx_e = np.zeros((ocp.dims.nx, ocp.dims.nx))
+
+        ocp.cost.yref = np.zeros((ocp.dims.ny,))
+        ocp.cost.yref_e = np.ones((ocp.dims.ny_e,))
+
+        # set constraints
+        ocp.constraints.x0 = np.array(self.nlp[0]["X_bounds"].min[:, 0])
+        ocp.dims.nbx_0 = ocp.dims.nx
+        ocp.constraints.constr_type = 'BGH'  # TODO: put as an option
+        ocp.constraints.lbu = np.array(self.nlp[0]["U_bounds"].min[:, 0])
+        ocp.constraints.ubu = np.array(self.nlp[0]["U_bounds"].max[:, 0])
+        ocp.constraints.idxbu = np.array(range(ocp.dims.nu))
+        ocp.dims.nbu = ocp.dims.nu
+
+        # set control constraints
+        ocp.constraints.Jbx_e = np.eye(ocp.dims.nx)
+        ocp.constraints.ubx_e = np.array(self.nlp[0]["X_bounds"].max[:, -1])
+        ocp.constraints.lbx_e = np.array(self.nlp[0]["X_bounds"].min[:, -1])
+        ocp.constraints.idxbx_e = np.array(range(ocp.dims.nx))
+        ocp.dims.nbx_e = ocp.dims.nx
+
+        ocp.solver_options.qp_solver = 'PARTIAL_CONDENSING_HPIPM'  # FULL_CONDENSING_QPOASES
+        ocp.solver_options.hessian_approx = 'GAUSS_NEWTON'
+        ocp.solver_options.integrator_type = 'ERK'
+        ocp.solver_options.nlp_solver_type = 'SQP'
+
+        ocp.solver_options.nlp_solver_tol_comp = 1e-02
+        ocp.solver_options.nlp_solver_tol_eq = 1e-02
+        ocp.solver_options.nlp_solver_tol_ineq = 1e-02
+        ocp.solver_options.nlp_solver_tol_stat = 1e-02
+        ocp.solver_options.sim_method_newton_iter = 5
+        ocp.solver_options.sim_method_num_stages = 4
+        ocp.solver_options.sim_method_num_steps = 10
+        ocp.solver_options.print_level = 1
```

Comment:
These should be in the solve function and allowed to change the same way option_ipopt (maybe changing the name for option_solver) is doing